haystack/components/generators/chat/hugging_face_local.py

```diff
@@ -0,0 +1,366 @@
+import logging
+import sys
+from typing import Any, Dict, List, Literal, Optional, Union, Callable
+
+from haystack.components.generators.hf_utils import PIPELINE_SUPPORTED_TASKS
+
+from haystack import component, default_to_dict, default_from_dict
+from haystack.components.generators.hf_utils import HFTokenStreamingHandler
+from haystack.components.generators.utils import serialize_callback_handler, deserialize_callback_handler
+from haystack.dataclasses import ChatMessage, StreamingChunk
+from haystack.lazy_imports import LazyImport
+
+logger = logging.getLogger(__name__)
+
+with LazyImport(message="Run 'pip install transformers[torch]'") as torch_and_transformers_import:
+    import torch
+    from huggingface_hub import model_info
+    from transformers import StoppingCriteriaList, pipeline, PreTrainedTokenizer, PreTrainedTokenizerFast
+    from haystack.components.generators.hf_utils import StopWordsCriteria  # pylint: disable=ungrouped-imports
+
+
+@component
+class HuggingFaceLocalChatGenerator:
+    """
+
+    The `HuggingFaceLocalChatGenerator` class is a component designed for generating chat responses using models from
+    Hugging Face's model hub. It is tailored for local runtime text generation tasks and provides a convenient interface
+    for working with chat-based models, such as `mistralai/Mistral-7B-Instruct-v0.2` or `meta-llama/Llama-2-7b-chat-hf`
```

Comment:
I thought of replacing mistralai/Mistral-7B-Instruct-v0.2 with a proper chat model (such as HuggingFaceH4/zephyr-7b-beta), even if Mistral supports the Chat Templating...
WDYT?